specification/intelligentdataexperiences/resource-manager/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/preview/2023-03-01-preview/aadcustomsecurityattributesdiagnosticsettings.json

```diff
@@ -0,0 +1,444 @@
+        },
+        "logs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LogSettings"
+          },
+          "description": "The list of logs settings.",
+          "x-ms-identifiers": [
+            "category"
+          ]
+        }
+      },
+      "description": "The diagnostic settings.",
+      "type": "object"
+    },
+    "DiagnosticSettingsResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DiagnosticSettings",
+          "description": "Properties of a Diagnostic Settings Resource."
+        }
+      },
+      "description": "The diagnostic setting resource."
+    },
+    "DiagnosticSettingsResourceCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiagnosticSettingsResource"
+          },
+          "description": "The collection of diagnostic settings resources."
+        }
+      },
+      "description": "Represents a collection of alert rule resources.",
+      "type": "object"
+    },
+    "DiagnosticSettingsCategory": {
+      "properties": {
+        "categoryType": {
+          "enum": [
+            "Logs"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "categoryType",
+            "modelAsString": true
+          },
+          "x-nullable": false,
+          "description": "The type of the diagnostic settings category."
+        }
+      },
+      "description": "The diagnostic settings Category.",
+      "type": "object"
+    },
+    "DiagnosticSettingsCategoryResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiagnosticSettingsCategory",
+          "description": "The properties of a Diagnostic Settings Category."
+        }
+      },
+      "description": "The diagnostic settings category resource."
+    },
+    "DiagnosticSettingsCategoryResourceCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiagnosticSettingsCategoryResource"
+          },
+          "description": "The collection of diagnostic settings category resources."
+        }
+      },
+      "description": "Represents a collection of diagnostic setting category resources.",
+      "type": "object"
+    }
+  },
+  "parameters": {
+    "DiagnosticSettingName": {
+      "name": "diagnosticSettingName",
+      "in": "path",
+      "required": true,
```

Comment:
The resource name parameter 'diagnosticSettingName' should be defined with a 'pattern' restriction.

See LintDiff error: ResourceNameRestriction.